semana10/labe-x/src/screens/CreateTripPage/CreateTripPage.jsx

```diff
@@ -0,0 +1,103 @@
+import Axios from 'axios';
+import React, { useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { baseURL } from '../../constans';
+import useForm from '../../hooks/useForm';
+import { goBack, goToAdminListTripsPage, goToHomePage, goToLoginPage } from '../../router/goToPages'
+import { MainContainer, ButtonWrapper, StandardButton, StandardForm, InputWrapper, ShortTextInput, LongTextInput, SelectInput, DateInput, NumberInput, StandardLabel, SecondaryButton  } from '../../styles'
+
+const CreateTripPage = () => {
+    const history = useHistory();
+
+    const {form, onChange, resetState} = useForm({
+      tripName: "",
+      planet: "",
+      tripDate: "",
+      tripDescription: "",
+      tripDuration: 0
+  })
+
+    useEffect(() => {
+      const token = localStorage.getItem("token");
+  
+      if (!token) {
+        goToLoginPage(history)
+      }
+  }, [history]);
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    onChange(name, value)
+  }
+
+    const postTrip = () => {
+      const body = {
+        name: form.tripName,
+        planet: form.planet,
+        date: form.tripDate,
+        description: form.tripDescription,
+        durationInDays: form.tripDuration
+      }
+      
+      Axios.post(`${baseURL}/trips`, body, 
+      {
+        headers : {
+          auth: localStorage.getItem("token")
+        }}).then(response => {
+          alert("Viagem cadastrada com sucesso")
+        }).then(
+          goToAdminListTripsPage(history)
+        ).catch(error => console.log(error))
```

Comment:
Acho que isso deve ter passado batido, mas tem um `then` a mais aí...

Reply:
A gente pode usar mais de um `then` quando queremos que uma promise seja executada depois da outra. Nesse caso aí, eu queria garantir que daria tempo de aparecer o alert antes do redirecionamento da página. Mas quando eu for revisar o código todo vou ver se realmente precisa disso. Talvez funcione mesmo colocando tudo no mesmo then :)